app/controllers/requests.rb

```diff
@@ -0,0 +1,53 @@
+class MakersBnb < Sinatra::Base
+  post '/request/new' do
+    space = Space.get(params[:space_id])
+    validate(space) 
+    request = Request.create(date_from: params[:date_from],
+                             date_to: params[:date_to],
+                             status: "Not Confirmed",
+                             user_id: current_user.id, 
+                             space_id: params[:space_id])
+    if request.saved?
+      flash.next[:saved] = ['Your booking request has been sent']
+      redirect('/spaces')
+    end
+  end
+
+  get '/requests' do
+    @spaces = Space.all(user_id: current_user.id) 
+    @requests = @spaces.requests
+    @space_names =  []
+    @requests.each do |request|
+      @space_names << Space.get(request.space_id).name
+    end
+    erb :'requests/requests'
```

Comment:
I don't think you need the instance var `@space_names`. You can just fetch them from `request.space`, no?